### PR TITLE
chore: remove useless type assertion

### PR DESCRIPTION
**Pull request by danranVm**
**Description:**
add generics to help type inference

### CODE REVIEW

File: src/internal/util/isPromise.ts [hunk 1]
```diff
@@ -3,6 +3,6 @@
  * @see {@link https://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects}
  * @param value the object to test
  */
-export function isPromise(value: any): value is PromiseLike<any> {
-  return !!value && typeof (<any>value).subscribe !== 'function' && typeof (value as any).then === 'function';
+export function isPromise<T = any>(value: any): value is PromiseLike<T> {
+  return !!value && typeof value.subscribe !== 'function' && typeof value.then === 'function';
```
Comment (cartant) [inline src/internal/util/isPromise.ts]: IMO, this change is unnecessary and is incorrect. Essentially, it uses a type parameter as a type assertion and I think that should be avoid. The current types accurately describe what this user-defined type-guard function does.

File: src/internal/util/subscribeTo.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export const subscribeTo = <T>(result: ObservableInput<T>): (subscriber: Subscri
   } else if (isArrayLike(result)) {
     return subscribeToArray(result);
   } else if (isPromise(result)) {
-    return subscribeToPromise(result as Promise<any>);
+    return subscribeToPromise(result);
```
Comment (cartant) [inline src/internal/util/subscribeTo.ts]: This change is fine, though. The type assertion in the current codebase is unnecessary and can be removed.

File: src/internal/util/isPromise.ts (hunk 1 above)
Comment (danranVm) [reply src/internal/util/isPromise.ts]: - this situation cannot be deduced correctly
image

- I hope so
image

Comment (cartant) [reply src/internal/util/isPromise.ts]: If `isPromise` is used in that way in this package, the usage should be changed - `Promise<string> | any` is equivalent to `any`, so there is no inference that can be drawn from that type.

And `isPromise` is an internal function, so it should not use used outside of this package - it is not exported, AFAICT.

And the the snippet you have above illustrates my point. It isn't inferring anything; it's a type parameter acting as a type assertion. IMO, it is better to be explicit with a type assertion.

Comment (danranVm) [reply src/internal/util/isPromise.ts]: Okay, I didn't notice `isPromise` is an internal function.

**Pull request merged**